src/Middleware/OutputCaching/src/OutputCacheAttribute.cs

```diff
@@ -45,9 +45,14 @@ public bool NoStore
     public string[]? VaryByQueryKeys { get; init; }
 
     /// <summary>
-    /// Gets or sets the headers to vary by.
+    /// Gets or sets the header names to vary by.
     /// </summary>
-    public string[]? VaryByHeaders { get; init; }
+    public string[]? VaryByHeaderNames { get; init; }
```

Comment:
I just saw your comment on the issue about this. I'm okay with this rename (that's why I liked renaming the `header` parameter to `headerNames`), but we should update the API proposal.

It is sad that we can make clarify the `VaryByHeaderNames` naming, but `VaryByRouteValues` still implies that it's a `string[]` of route values and not route value names.

Do we want to change all our references to `VaryByRouteValues` to be `VaryByRouteValueNames` too? I feel that's the only what to be consistent.

Otherwise, we're probably just better of making them `VaryByQuery`, `VeryByHeaders` and `VaryByRouteValues`.

The current naming makes it seem like some of these are names and some of these are values. At least that way we're consistently unclear and we can provide more detail with doc comments.